### ISSUE TITLE
Need to clarify docs about a hidden GitHub pages requirement on private repository

### ISSUE THREAD

**Issue by huy-nguyen** [peaceiris/actions-gh-pages#349]
### Is your feature request related to a problem? Please describe

I created a new repo and use this action to publish to the `gh-pages` branch but the repo's supposed GitHub page returned a 404 even after choosing `gh-pages` branch from the dropdown in the repo's GitHub pages setting. 

I contacted GitHub support about this and their response was 

> In order to be built and published for the very first time, a GitHub Pages repository must have a commit pushed to it by a user with admin permissions for the repository. In this case, it looks like while you committed to the master branch, the only commit pushed to the gh-pages branch was from `github-actions`.

Following this tip fixed my problem.

### Describe the solution you'd like

Add an explanation to the docs. The easiest solution would be for the user to push any content under their *own* account to the `gh-pages` branch and after that this action can take over the publishing.
### Describe alternatives you've considered

There's no alternatives. This seems to be a GitHub requirement.

**Comment by peaceiris** [comment]
Thank you for suggesting this!

Now, we have the explanation about this in the [First Deployment with GITHUB_TOKEN](https://github.com/peaceiris/actions-gh-pages#%EF%B8%8F-first-deployment-with-github_token) section.

> The GITHUB_TOKEN has limitations for the first deployment so we have to select the GitHub Pages branch on the repository settings tab. After that, do the second deployment like the following pictures.

Should we put it on the top of the README?

**Comment by huy-nguyen** [comment]
What I did was running the action, choose the gh-pages branch, run the action again, which is essentially your "doing 2 deployments" solution but that didn't fix the problem. I think the problem here was that the github action is not considered "a user with admin permissions." Maybe you can do some testing on your own to validate my findings?

**Comment by peaceiris** [comment]
> What I did was running the action, choose the gh-pages branch, run the action again, which is essentially your "doing 2 deployments" solution but that didn't fix the problem.

When we have no change in deployment assets, we get the message `nothing to commit, working tree clean` and deployment will be skipped. Was your second deployment skipped? Please check your log [like this](https://github.com/peaceiris/actions-gh-pages/commits/gh-pages).

![GitHub Pages log](https://raw.githubusercontent.com/peaceiris/actions-gh-pages/master/images/log_success.jpg)

>  I think the problem here was that the github action is not considered "a user with admin permissions." Maybe you can do some testing on your own to validate my findings?

Yes. This is a problem we have been facing for several months. You can see the first discussion and investigation of deployment with GITHUB_TOKEN at #9. Some of the problems have been solved and others (this issue!) are still opened today. This is an inside authentication problem of GitHub Pages and GitHub service. Currently, we have no solution to this, (only GitHub developer team can work on this) I think.

**Comment by peaceiris** [comment]
I have tested this at [peaceiris/test-docusaurus](https://github.com/peaceiris/test-docusaurus) now. It works well for me.

1. Delete the gh-pages branch for testing
1. Do the first deployment
1. Click the gh-pages branch button in the settings tab.
1. Do the second deployment
1. I got the following log (GitHub Pages enabled successfully)

<img width="667" alt="GitHub Pages deployment log with GitHub Actions" src="https://user-images.githubusercontent.com/30958501/84582000-5ca9b100-ae21-11ea-88d3-8fb72377fdb5.png">

**Comment by letmaik** [comment]
I had the same issue. On a new repository, I followed the two-step instructions but it was impossible to get it deployed. The only way out was to manually push a commit to `gh-pages`. And after that the automatic deploys worked.

**Comment by huy-nguyen** [comment]
@letmaik that was what I was trying to tell the maintainer.

**Comment by peaceiris** [comment]
OK. @huy-nguyen and @letmaik
Could you share your public repository for reproducing?

Now, I deleted [peaceiris/test-docusaurus](https://github.com/peaceiris/test-docusaurus) and deployed it with the 2-step instructions. It works well. I could not get a failure case on my side.

<img width="611" alt="Screen Shot 2020-06-21 at 13 55 02" src="https://user-images.githubusercontent.com/30958501/85217053-71f58100-b3c7-11ea-8a84-00dea90e701a.png">

**Comment by peaceiris** [comment]
I got it. Your problems are related to a private repository, right?

I could reproduce it on my side.

<img width="1034" alt="Screen Shot 2020-06-21 at 14 44 22" src="https://user-images.githubusercontent.com/30958501/85217638-f8ad5c80-b3cd-11ea-8982-08e95fa8ce3f.png">

Now, we have no solution on the actions-gh-pages side. I am investigating it.

Anyway, we need to announce this case on our README.

Thank you!

**Comment by letmaik** [comment]
@peaceiris Yes, it's a private one.

**Comment by huy-nguyen** [comment]
Yes this happens for private repos because since GitHub changed their pricing structure recently, only paid accounts can have GitHub pages for private repos.

**Comment by sl4m** [comment]
Glad I found this issue. I was having trouble with a private repo myself and found that pushing a manual commit to `gh-pages` branch solved the problem as suggested

**Comment by chatziko** [comment]
I'm also affected by this and I did lots of tests. It seems to me that on a private repo, __none__ of the 3 authentication methods currently works, I tried github_token, deploy_key and  personal_token and get the same behaviour:
- the action runs fine
- the master branch is updated
- gh-pages log says deployed
- but serves the old version.

I even tried a personal token + manually asking for a build:
```
    - name: Deploy
      uses: peaceiris/actions-gh-pages@v3
      with:
        personal_token: ${{ secrets.PERSONAL_TOKEN }}
        publish_branch: master
        publish_dir: ./public
        force_orphan: true

    - name: Request Github Pages build job
      run: >-
        curl -L -X POST
        -H "Content-Type: application/json"
        -H "Authorization: token ${{ secrets.PERSONAL_TOKEN }}"
        "https://api.github.com/repos/${{ github.repository }}/pages/builds"
```
Same thing.

I also tried a __personal__ ssh key (configured in my personal settings, not a deploy key), even that does not work.

At the same time, manually pushing an empty commit to the master branch updates the served page within a few seconds.

I guess something is seriously wrong on the guthub side, I hope they fix it soon.
